### DISCUSSION TITLE
How to get command parameters when the command is queued

### DISCUSSION THREAD

**Discussion by melvinotieno** [laravel/framework#51294] [Q&A] [answered]
Given the following command

```php
Artisan::queue('mail:send', [
    'user' => $user, '--queue' => 'default'
]);
```

I am listening to when the Queue ends through

```php
Queue::after(function (JobProcessed $event) {
    $job = $event->job;
    $data = $job->payload()['data'];

    if ($job->resolveName() === 'mail:send') {
        $command = unserialize($data['command']);

        // how do i get the command parameters from here?
    }
}
```

How do I get the command parameters `user` and `--queue` passed to the Artisan command queue?

**Comment by ghost** [comment]
Mock a command locally with queue driver database and workers STOPPED and look in db to see the command key-value.

**Reply by melvinotieno** [reply]
I am not sure I got what you mean. Also, I am using redis as the queue driver

**Reply by ghost** [reply]
Well use database locally. The payload is the same as it is for Redis.
Add a job/message to your local db queue. Then read from it to see the format.

**Reply by melvinotieno** [reply]
Ooh, I already know the format. It comes back as QueuedCommand. When I check the class, it does have the data but it is protected so I cannot get it

**Reply by ghost** [reply]
You can use reflection to read protected properties from a class.

**Reply by melvinotieno** [reply]
Oh yeah, that didn't cross my mind. Thank you

**Comment by ghost** [comment, marked as answer]
You can use reflection to read protected properties from a class.

**Reply by henzeb** [reply]
Or use closures for that. Less code required ...

**Reply by ghost** [reply]
@henzeb You made me curious. How can it be done with closures?

**Reply by henzeb** [reply]
Using https://www.php.net/manual/en/closure.bind.php you can bind the closure to the object with a different context e.g. the parent class. you then have access to protected, and I believe even private, properties and methods.
